use-cases/clickstream-analytics-dashboard/dashboard.py

```diff
@@ -0,0 +1,113 @@
+import glassflow
+import sys
+from dotenv import load_dotenv
+import streamlit as st
+import plotly.express as px
+import pandas as pd
+import os
+import time
+
+load_dotenv()
+
+pipeline_id = os.environ["PIPELINE_ID"]
+space_id = os.environ["SPACE_ID"]
+token = os.environ["PIPELINE_ACCESS_TOKEN"]
+client = glassflow.GlassFlowClient()
+pipeline_client = client.pipeline_client(
+    space_id=space_id, pipeline_id=pipeline_id, pipeline_access_token=token
+)
+
+st.title("Clickstream Analytics Dashboard")
+
+placeholder = st.empty()
+
+
+def update_dashboard(data):
+    if data:
+        df = pd.DataFrame(data)
+
```

Comment:
is the "data" a single row or not? why a dataframe for a single row?

Reply:
Ok, i see that "data" is a global variable of type list which is appended with new data and the dashboard is refreshed on every update.

Reply:
Yes, correct data rows are increased to keep previous events and update the dashboard with all events. I tried to simulate somehow stateful tables update:)